### PR TITLE
[S5-06] trang chu giang vien(Thống kê)

### PR DESCRIPTION
**Pull request by duyminh256**

### CODE REVIEW

**Comment by nhoknguyen00** [comment]
Nên viết thêm seed để dễ test

File: src/catalog/instructor/instructorController.js [hunk 1]
```diff
@@ -28,3 +29,85 @@ export const getInstructorInfo = async (req, res, next) => {
...
+ * @param {*} req
+ * @param {*} res
+ * @param {*} next
+ */
+
+export const getIncomeAnalytics = async (req, res, next) => {
+  try {
+    const { id } = req.user
+    const { course, from, to, type } = req.body
+    const instructorId = await instructorService.getInstructorInfoByUserId(id)
```
Comment (hminh21) [inline src/catalog/instructor/instructorController.js]: `const { id: instructorId } = await instructorService.getInstructorInfoByUserId(id)`

File: src/catalog/instructor/instructorController.js [hunk 2]
```diff
@@ -28,3 +29,85 @@ export const getInstructorInfo = async (req, res, next) => {
...
+    res.status(httpStatusCodes.INTERNAL_SERVER_ERROR).json({
+      message: 'Hệ thống gặp lỗi khi lấy dữ liệu doanh thu'
+    })
+  }
+}
+export const getStudentNumberAnalytics = async (req, res, next) => {
+  try {
+    const { id } = req.user
+    const { course, from, to, type } = req.body
+    const instructorId = await instructorService.getInstructorInfoByUserId(id)
```
Comment (hminh21) [inline src/catalog/instructor/instructorController.js]: `const { id: instructorId } = await instructorService.getInstructorInfoByUserId(id)`

File: src/catalog/instructor/instructorController.js [hunk 3]
```diff
@@ -28,3 +29,85 @@ export const getInstructorInfo = async (req, res, next) => {
...
+    res.status(httpStatusCodes.INTERNAL_SERVER_ERROR).json({
+      message: 'Hệ thống gặp lỗi khi lấy dữ liệu số lượng học viên'
+    })
+  }
+}
+export const getActiveCourseAnalytics = async (req, res, next) => {
+  try {
+    const { id } = req.user
+    const { course, from, to, type } = req.body
+    const instructorId = await instructorService.getInstructorInfoByUserId(id)
```
Comment (hminh21) [inline src/catalog/instructor/instructorController.js]: `const { id: instructorId } = await instructorService.getInstructorInfoByUserId(id)`

File: src/catalog/instructor/instructorController.js [hunk 4]
```diff
@@ -28,3 +29,85 @@ export const getInstructorInfo = async (req, res, next) => {
...
+    debug.error(NAMESPACE, 'Error occured while getting data active course analytics', err)
+    res.status(httpStatusCodes.INTERNAL_SERVER_ERROR).json({
+      message: 'Hệ thống gặp lỗi khi lấy dữ liệu khóa học sôi động'
+    })
+  }
+}
+export const getCourses = async (req, res, next) => {
+  try {
+    const { id } = req.user
+    const instructorId = await instructorService.getInstructorInfoByUserId(id)
```
Comment (hminh21) [inline src/catalog/instructor/instructorController.js]: `const { id: instructorId } = await instructorService.getInstructorInfoByUserId(id)`

File: src/catalog/instructor/instructorRoute.js [hunk 5]
```diff
@@ -10,4 +10,9 @@ const router = express.Router()
  */
 router.get('/detail/:id', instructorController.getInstructorInfo)
 
+router.get('/course', auth(), instructorController.getCourses);
```
Comment (hminh21) [inline src/catalog/instructor/instructorRoute.js]: `router.get('/get-course', auth({ isInstructor }), instructorController.getCourses)`

File: src/catalog/instructor/instructorRoute.js [hunk 6]
```diff
@@ -10,4 +10,9 @@ const router = express.Router()
...
+router.post('/analytics/income', auth(), instructorController.getIncomeAnalytics)
```
Comment (hminh21) [inline src/catalog/instructor/instructorRoute.js]: `router.post('/analytics/income', auth({ isInstructor }), instructorController.getIncomeAnalytics)`

File: src/catalog/instructor/instructorRoute.js [hunk 7]
```diff
@@ -10,4 +10,9 @@ const router = express.Router()
...
+router.post('/analytics/student-number', auth(), instructorController.getStudentNumberAnalytics)
```
Comment (hminh21) [inline src/catalog/instructor/instructorRoute.js]: `router.post('/analytics/student-number', auth({ isInstructor }), instructorController.getStudentNumberAnalytics)`

File: src/catalog/instructor/instructorRoute.js [hunk 8]
```diff
@@ -10,4 +10,9 @@ const router = express.Router()
...
+router.post('/analytics/active-course', auth(), instructorController.getActiveCourseAnalytics)
```
Comment (hminh21) [inline src/catalog/instructor/instructorRoute.js]: `router.post('/analytics/active-course', auth({ isInstructor }), instructorController.getActiveCourseAnalytics)`

**Pull request merged**